### PR TITLE
Implementing `reconfig`, `sync`, `exists` commands for keeper-client

### PR DESCRIPTION
**Pull request by pufit**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implementation of `reconfig` (https://github.com/ClickHouse/ClickHouse/pull/49450), `sync`, and `exists` commands for keeper-client. 

https://github.com/ClickHouse/ClickHouse/issues/54129

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 096b44e1e5dcf46e081b9d8a123462c222c8fd0b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54201/096b44e1e5dcf46e081b9d8a123462c222c8fd0b/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by antonio2368 — APPROVED** [review]
LGTM but we really need tests
something like we did for the client in `tests/integration/helpers/client.py` we should be able to do with `keeper-client`

File: programs/keeper-client/KeeperClient.cpp [hunk 1]
```diff
@@ -313,6 +330,26 @@ void KeeperClient::runInteractive()
...
 }
 
+void KeeperClient::runInteractiveInputStream()
+{
+    for (String input; std::getline(std::cin, input);)
+    {
+        if (!processQueryText(input))
+            break;
+
+        std::cout << "\a\a\a\a" << std::endl;
```
Comment (pufit) [inline programs/keeper-client/KeeperClient.cpp:340]: This is a hack that hopefully will be removed after supporting JSON output format.

File: tests/integration/helpers/keeper_utils.py [hunk 2]
```diff
@@ -1,6 +1,183 @@
...
+    )
+
+    return request.get_answer()
+
+
+class KeeperException(Exception):
+    pass
+
+
+class KeeperClient(object):
```
Comment (antonio2368) [inline tests/integration/helpers/keeper_utils.py:37]: ❤️

File: programs/keeper-client/Commands.h [hunk 3]
```diff
@@ -177,6 +188,35 @@ class RMRCommand : public IKeeperClientCommand
     String getHelpMessage() const override { return "{} <path> -- Recursively deletes path. Confirmation required"; }
 };
 
+class ReconfigCommand : public IKeeperClientCommand
+{
+    enum class Operation : Int64
```
Comment (antonio2368) [inline programs/keeper-client/Commands.h:193]:
```suggestion
    enum class Operation : UInt8
```

File: programs/keeper-client/Commands.cpp [hunk 4]
```diff
@@ -355,6 +372,64 @@ void RMRCommand::execute(const ASTKeeperQuery * query, KeeperClient * client) co
...
+
+    auto operation = query->args[0].get<ReconfigCommand::Operation>();
+    if (operation == static_cast<Int64>(ReconfigCommand::Operation::ADD))
+        joining = query->args[1].safeGet<DB::String>();
+    else if (operation == static_cast<Int64>(ReconfigCommand::Operation::REMOVE))
+        leaving = query->args[1].safeGet<DB::String>();
+    else if (operation == static_cast<Int64>(ReconfigCommand::Operation::SET))
+        new_members = query->args[1].safeGet<DB::String>();
+    else
+        UNREACHABLE();
```
Comment (antonio2368) [inline programs/keeper-client/Commands.cpp:412]: switch statements?

File: tests/integration/test_keeper_reconfig_remove/test.py [hunk 5]
```diff
@@ -42,46 +44,43 @@ def test_reconfig_remove_followers_from_3(started_cluster):
     Check that remaining node is in standalone mode.
     """
 
-    zk1 = get_fake_zk(node1)
-    config, _ = zk1.get("/keeper/config")
-    config = config.decode("utf-8")
+    global zk1, zk2, zk3
```
Comment (antonio2368) [inline tests/integration/test_keeper_reconfig_remove/test.py:47]: why?

Comment (pufit) [reply tests/integration/test_keeper_reconfig_remove/test.py:47]: To have a graceful shutdown https://github.com/ClickHouse/ClickHouse/pull/54201/files#diff-dbf5606f605734e2f5d6335548b5f91e728fb9084255e3d397dee333ac792679R34

**Comment by pufit** [comment]
The new client in tests seems to be flaky, investigating it.

**Pull request merged**